### ISSUE TITLE
Targeted rewrite: conv_rw?

### ISSUE THREAD

**Issue by siddhartha-gadgil** [siddhartha-gadgil/LeanAide#27]
This follows an idea of Scott Morrison, and is for use both directly in Mathlib (potentially) and with LeanAide.

* Instead of matching `DiscTree`'s in all subexpressions, one navigates into lambdas and function applications as in the `conv` tactic mode.
* The resulting candidates are of the form `conv => enter[...] => rw [..]`

**Comment by 0art0** [comment]
The built-in Lean type [`SubExpr`](https://github.com/0art0/lean4/blob/master/src/Lean/SubExpr.lean) may be useful here. A `SubExpr` is determined by a `SubExpr.Pos`, which is a type synonym for `Nat`; however, the function `SubExpr.Pos.toArray` decodes the `Nat` to a familiar `Array` of indices which determine the path along the expression tree.

**Comment by 0art0** [comment]
Closely related to targeted rewriting with `conv` [is this demo](https://github.com/EdAyers/ProofWidgets4/blob/cf2e6e09166e1e8d791f8435bf1e3769da2d5b86/ProofWidgets/Demos/Conv.lean) frpm the `ProofWidgets` repository. A variation of the `solveLevel` function might be what is required.

**Comment by siddhartha-gadgil** [comment]
@0art0 Does the widget work under binders, for example with:

```lean
example : ∀ x: ℕ, x + 3 = 3 + x := by
  conv =>
    enter [x, 2]
    rw [Nat.add_comm]
  intro _ 
  rfl
```

If so I agree we should follow that code, with navigating to all positions instead of based on clicks.

**Comment by 0art0** [comment]
Yes, I tested the above example and it does work under binders too.

**Comment by siddhartha-gadgil** [comment]
@0art0 What is the suggested tactic sequence there? Does it give `enter [x, 2]` or equivalent?

**Comment by 0art0** [comment]
Yes, it gave `enter [x, 1]` (as I chose to rewrite the left-hand side).

**Comment by siddhartha-gadgil** [comment]
That is perfect then. One just has to copy part of the code (excluding widget stuff) and simulate clicking everywhere. Then one can use the code of `rw?` (maybe not going into sub-expressions).

**Comment by siddhartha-gadgil** [comment]
We are not trying to improve automation. So this will not be a priority.
